app/src/test/java/waiter/ClientConnectionTest.java

```diff
@@ -0,0 +1,35 @@
+package waiter;
+
+import waiter.ClientConnection.ClientConnection;
+
+import org.junit.jupiter.api.Test;
+import waiter.InputStreamer.mock.InputStreamerMock;
+import waiter.OutputStreamer.mock.OutputStreamerMock;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.io.IOException;
+
+public class ClientConnectionTest {
+
+    @Test
+    void readsMessage() throws IOException {
+        String input = "foo";
+        InputStreamerMock inputStreamerMock = new InputStreamerMock(input);
+        OutputStreamerMock outputStreamerMock = new OutputStreamerMock();
+        ClientConnection clientConnection = new ClientConnection(inputStreamerMock, outputStreamerMock);
```

Comment:
What do you think about inlining this:
```java
String input = "foo";

ClientConnection clientConnection = new ClientConnection(
  new InputStreamerMock(input),
  new OutputStreamerMock()
);
```

Reply:
Hmmm, I think this is a good idea, it doesn't really hurt readability and it makes it more concise, should I reflect this in `Listener` as well?

Reply:
Yea sounds good to me!